pyproject.toml

```diff
@@ -4,6 +4,7 @@ version = "0.1.0"
 requires-python = ">=3.8,<3.12"
 description = "Modalities, a python framework for distributed and reproducible foundation model training."
 dependencies = [
+    "numpy<2.0",
```

Comment:
why do we need the nump version to be `<2.0` ?

Reply:
Numpy 2.0 introduces breaking changes and is currently not compatible with the latest version of at least one of our dependencies (`accelerate`). It leads to the following error: 

```AttributeError: `np.float_` was removed in the NumPy 2.0 release. Use `np.float64` instead.. Did you mean: 'float16'?```